### PR TITLE
MAILBOX-355 MailboxListeners should not use MailboxPath

### PR DESCRIPTION
**Pull request by chibenwa**
Design choices:

 - Prefer using MailboxId instead of MailboxPath in logs (IMAP)
 - Allow retrieving QuotaRoot from MailboxId
 - Move SystemMailboxesProvider in Mailbox-Api
 - Remove Event Serialization code - Our future scala code gonna rock!

Still, remaining issues to be discussed: `mailbox/cache` makes use of MailboxPath in events for cache invalidation ...
 - Should we drop this? (it is a non exposed feature - without tests)
 - Should we handle that by mailboxId
 - Or do we keep MailboxPath in events, and deprecate it?

Please review. I can split this into parts at will, depending on what is controversial or not!

### CODE REVIEW

File: protocols/imap/src/main/java/org/apache/james/imap/processor/base/SelectedMailboxImpl.java [hunk 1]
```diff
@@ -162,8 +161,8 @@ public synchronized int recentCount() {
     }
 
     @Override
-    public synchronized MailboxPath getPath() {
-        return path;
+    public synchronized MailboxPath getPath() throws MailboxException {
```
Comment (mbaechler) [inline protocols/imap/src/main/java/org/apache/james/imap/processor/base/SelectedMailboxImpl.java]: does it still require to be `synchronized` ?

File: server/container/guice/protocols/jmap/pom.xml [hunk 2]
```diff
@@ -41,6 +41,10 @@
             <artifactId>apache-mailet-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>apache-james-mailbox-store</artifactId>
```
Comment (mbaechler) [inline server/container/guice/protocols/jmap/pom.xml]: why ? It's not what the commit says

File: mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinListener.java [hunk 3]
```diff
@@ -94,6 +93,16 @@ public void event(Event event) {
         }
     }
 
+    private boolean isAppendedToInbox(EventFactory.AddedImpl addedEvent) {
+        try {
+            return systemMailboxesProvider.findMailbox(Role.INBOX, addedEvent.getSession())
```
Comment (mbaechler) [inline mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinListener.java]: `==` is probably not ok, it shoud be `.equals` ?

File: mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinListener.java [hunk 4]
```diff
@@ -129,11 +138,7 @@ boolean isMessageMovedOutOfSpamMailbox(MessageMoveEvent event) {
     }
 
     private MailboxId getMailboxId(MessageMoveEvent event, Role role) throws MailboxException {
```
Comment (mbaechler) [inline mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinListener.java]: this method doesn't make much sense anymore

File: protocols/imap/src/main/java/org/apache/james/imap/processor/base/SelectedMailboxImpl.java (hunk 1 above)
Comment (chibenwa) [reply protocols/imap/src/main/java/org/apache/james/imap/processor/base/SelectedMailboxImpl.java]: No

File: server/container/guice/protocols/jmap/pom.xml (hunk 2 above)
Comment (chibenwa) [reply server/container/guice/protocols/jmap/pom.xml]: The binding for the systemMailboxprovider impl is node here, I will move it to a more generic module

File: mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinListener.java (hunk 3 above)
Comment (chibenwa) [reply mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinListener.java]: Good catch

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**